elenbaskakova-DOCSUP-780 (#87)
BayoNet/ClickHouse: docs/en/operations/system_tables.md

## system.detached_parts {#system_tables-detached_parts}

Contains information about detached parts of [MergeTree](table_engines/mergetree.md) tables. The `reason` column specifies why the part was detached. For user-detached parts, the reason is empty. Such parts can be attached with [ALTER TABLE ATTACH PARTITION|PART](../query_language/alter.md#alter_attach-partition) command. For the description of other columns, see [system.parts](#system_tables-parts). If part name is invalid, values of some columns may be `NULL`. Such parts can be deleted with [ALTER TABLE DROP DETACHED PART](../query_language/alter.md#alter_drop-detached).

## system.dictionaries

Contains information about [external dictionaries](../query_language/dicts/external_dicts.md).

Columns:

> - `database` (String) — Database name where the dictionary is located. Only for DDL dictionaries, for others is always an empty string.

Reviewer: Links to types.
DDL dictionary - > dictionary created by DDL query.

- `name` (String) — [Dictionary name](../query_language/dicts/external_dicts_dict.md).

> - `status` (Enum8) — Vocabulary status. Possible values: 

Reviewer: ```suggestion
- `status` (Enum8) — Dictionary status. Possible values: 
```

>      - `NOT_LOADED` — Dictionary did not load because it was not used.

Reviewer: ```suggestion
     - `NOT_LOADED` — Dictionary did not loaded because it was not used.
```

Author: 2 possible options:
1.  Dictionary did not load because it was not used.
2.  Dictionary was not loaded because it was not used.

     - `LOADED` — Dictionary loaded successfully.
     - `FAILED` — Unable to load the dictionary as a result of an error.
     - `LOADING` — Dictionary is loading now.

>      - `LOADED_AND_RELOADING` — Dictionary is loaded successfully, and is being reloaded right now (frequent reasons: SYSTEM RELOADDICTIONARY request, timeout, dictionary config has changed).

Reviewer: SYSTEM RELOAD DICTIONARY - https://clickhouse.tech/docs/en/query_language/system/#query_language-system-reload-dictionary. Query not request.

     - `FAILED_AND_RELOADING` - Could not load the dictionary as a result of an error and is loading now.
- `origin` (String) — Path to the configuration file that describes the dictionary.

> - `type` (String) — Dictionary type: [flat](../query_language/dicts/external_dicts_dict_layout.md#flat), [hashed](../query_language/dicts/external_dicts_dict_layout.md#dicts-external_dicts_dict_layout-hashed), [cache](../query_language/dicts/external_dicts_dict_layout.md#cache).

Reviewer: ```suggestion
- `type` (String) — Type of a dictionary allocation. [Storing Dictionaries in Memory](../query_language/dicts/external_dicts_dict_layout.md)
```

- `key` — [Key type](../query_language/dicts/external_dicts_dict_structure.md#ext_dict_structure-key): Numeric Key (UInt64) or Сomposite key (String) — form "(type 1, type 2, ..., type n)".
- `attribute.names` (Array(String)) — Array of [attribute names](../query_language/dicts/external_dicts_dict_structure.md#ext_dict_structure-attributes) provided by the dictionary.
- `attribute.types` (Array(String)) — Corresponding array of [attribute types](../query_language/dicts/external_dicts_dict_structure.md#ext_dict_structure-attributes) that are provided by the dictionary.
- `bytes_allocated` (UInt64) — Amount of RAM allocated for the dictionary.

> - `query_count` (UInt64) — Number of requests since the dictionary was loaded or since the last successful reboot.

Reviewer: ```suggestion
- `query_count` (UInt64) — Number of queries since the dictionary was loaded or since the last successful reboot.
```

- `element_count` (UInt64) — Number of items stored in the dictionary.
- `load_factor` (Float64) — Percentage filled in the dictionary (for a hashed dictionary, the percentage filled in the hash table).
- `source` (String) — Text describing the [data source](../query_language/dicts/external_dicts_dict_sources.md) for the dictionary.
- `lifetime_min` (UInt64) — Minimum [lifetime](../query_language/dicts/external_dicts_dict_lifetime.md) of the dictionary in memory, after which ClickHouse tries to reload the dictionary (if `invalidate_query` is set, then only if it has changed). Set in seconds.
- `lifetime_max` (UInt64) — Maximum [lifetime](../query_language/dicts/external_dicts_dict_lifetime.md) of the dictionary in memory, after which ClickHouse tries to reload the dictionary (if `invalidate_query` is set, then only if it has changed). Set in seconds.
- `loading_start_time` (DateTime) — Start time for loading the dictionary.
- `last_successful_update_time` (DateTime) — End time for loading or updating the dictionary. Helps to monitor some troubles with external sources and investigate causes.

> - `loading_duration` (Float32) — Time taken to load the dictionary.

Reviewer: ```suggestion
- `loading_duration` (Float32) — Duration of a dictionary loading.
```

- `name` (String) — Dictionary name.
- `status` (Enum8) — Vocabulary status. Possible values: 
     - `NOT_LOADED` — Dictionary did not load because it was not used.
     - `LOADED` — Dictionary loaded successfully.
     - `FAILED` — Unable to load the dictionary as a result of an error.
     - `LOADING` — Dictionary is loading now.
     - `LOADED_AND_RELOADING` — Dictionary is loaded successfully, and is being reloaded right now (frequent reasons: SYSTEM RELOADDICTIONARY request, timeout, dictionary config has changed).
     - `FAILED_AND_RELOADING` - Could not load the dictionary as a result of an error and is loading now.
- `origin` (String) — Path to the configuration file that describes the dictionary.
- `type` (String) — Dictionary type: flat, hashed, cache.
- `key` — Key type: Numeric Key (UInt64) or Сomposite key (String) — form "(type 1, type 2, ..., type n)".
- `attribute.names` (Array(String)) — Array of attribute names provided by the dictionary.
- `attribute.types` (Array(String)) — Corresponding array of attribute types that are provided by the dictionary.

> - `bytes_allocated` (UInt64) — Amount of RAM the dictionary uses.

Reviewer: ```suggestion
- `bytes_allocated` (UInt64) — Amount of RAM allocated for the dictionary.
```
